api/src/opentrons/protocol_api/contexts.py

```
@@ -248,22 +249,32 @@ def load_module(
                 'hc_name': 'thermocycler',
                 'geo_name': 'semithermocycler'}
             }
+        try:
+            hc_mod_name = mod_map[module_name.lower()]['hc_name']
+            geo_mod_name = mod_map[module_name.lower()]['geo_name']
 
-        hc_mod_name = mod_map[module_name.lower()]['hc_name']
-        geo_mod_name = mod_map[module_name.lower()]['geo_name']
+            geometry = load_module(
+                geo_mod_name, self._deck_layout.position_for(location))
+        except KeyError:
+            print(f'Unsupported Module: {module_name}')
```

Couple things:
- This should be a `log.error` not a `print`
- If you put in an invalid name, like 'csahugehw' catching and swallowing the exception just means you'll get an `UnboundLocalError` on line 265 when you try to use `hc_mod_name`, which doesn't exist
- If you put in an invalid name, we _should_ raise an exception because that _is_ an error that should stop simulation